ramips: split Youku YK1 to YK-L1 and YK-L1c

DragonBluep opened the pull request:
Device specifications:
* Model: Youku YK-L1/L1c
* CPU: MT7620A
* RAM: 128 MiB
* Flash: 32 MiB (YK-L1)/ 16 MiB (YK-L1c)
* LAN: 2* 10M/100M Ports
* WAN: 1* 10M/100M Port
* USB: 1* USB2.0
* SD: 1* MicroSD socket
* UART: 1* TTL, Baudrate 57600

Descriptions:
  Previous supported device YOUKU yk1 is actually Youku YK-L1. Though they look
  really different, the only hardware difference between the two models is flash
  size, YK-L1 has 32 MiB flash but YK-L1c has 16MiB. It seems that YK-L1c can
  compatible with YK-L1's firmware but it's better to split it to different models.

  It is easy to identify the models by looking at the label on the bottom of the
  device. The label has the model number "YK-L1" or "YK-L1c". Due to different flash
  sizes, YK-L1c that using previous YK-L1's firmware needs to apply "force update"
  to install compatible firmware, so please backup config file before system upgrade.

adschm commented:
This has formal issues. Please squash your changes into one (or a few) proper commits.

https://openwrt.org/submitting-patches

adschm commented on target/linux/ramips/image/mt7620.mk:
```diff
@@ -1109,16 +1109,27 @@ define Device/xiaomi_miwifi-mini
   DEVICE_PACKAGES := kmod-usb2 kmod-usb-ohci kmod-sdhci-mt7620 \
 	kmod-usb-ledtrig-usbport
-  SUPPORTED_DEVICES += youku-yk1
+  SUPPORTED_DEVICES += yk-l1
```
What are the SUPPORTED_DEVICES changes supposed to do? Looks like you are not aware of what this is for?

DragonBluep replied:
I give up this change in this commit, one commit one change, I got it.

adschm commented on target/linux/ramips/mt7620/base-files/etc/board.d/02_network:
```diff
@@ -217,6 +216,11 @@ ramips_setup_interfaces()
+	youku,yk-l1|\
+	youku,yk-l1c)
+		ucidef_add_switch "switch0" \
+			"0:lan" "1:lan" "2:lan" "6@eth0"
```
If you intend to change the network config, please do that in a separate (proper) commit.

DragonBluep replied:
okay, I will obey the rules. I'll change the commit soon. But before that I really need to learn more git operate tips😂

DragonBluep commented:
I have re-written commit  and it looks okay now. Maybe we can remove "needs changes" label. @adschm

adschm commented:
What should be a DTSI is a DTS. So, I assume this has not been even build-tested in its current state?

DragonBluep commented:
> What should be a DTSI is a DTS. So, I assume this has not been even build-tested in its current state?

Sorry, I should convert this PR to draft because I created another related commit d09c2200fd21096be84a2e9f35ada76f7ca202f2 not long ago. And now everything be okay. The latest commit has been tested in my local host.

DragonBluep commented:
It seems that there are already some people flash Youku YK-L1's image to their YK-L1c, refer to [Bug 2717](https://bugs.openwrt.org/index.php?do=details&task_id=2717). Now I re-write the commit to split YK1 to different models instead of add support to a new device.

mpratt14 commented:
are the 2 devices have the same FCC ID or different? this would be good to include in commit message, to help identify differences.

adschm commented on target/linux/ramips/dts/mt7620a_youku.dtsi:
```diff
@@ -88,10 +85,9 @@
+			firmware: partition@50000 {
 				compatible = "denx,uimage";
 				label = "firmware";
-				reg = <0x50000 0x1fb0000>;
```
Please leave a comment like the following instead of the reg property here:
`/* reg property is set based on flash size in DTS files */`

DragonBluep replied:
Now added, thx.

adschm commented on target/linux/ramips/image/mt7620.mk:
```diff
@@ -1109,16 +1109,27 @@ define Device/xiaomi_miwifi-mini
+  DEVICE_MODEL := YK-L1c
+  DEVICE_PACKAGES := kmod-usb2 kmod-usb-ohci kmod-sdhci-mt7620 \
+	kmod-usb-ledtrig-usbport
+  SUPPORTED_DEVICES += youku-yk1 youku,yk1
```
Upgrade path should only be available for the same flash size. Please drop the additional supported devices for the L1C variant here.

adschm replied:
... and please also add a short comment about this fact to the commit message.

adschm commented:
This looks generally fine ...

DragonBluep committed:
ramips: split Youku YK1 to YK-L1 and YK-L1c

Device specifications:
* Model: Youku YK-L1/L1c
* CPU: MT7620A
* RAM: 128 MiB
* Flash: 32 MiB (YK-L1)/ 16 MiB (YK-L1c)
* LAN: 2* 10M/100M Ports
* WAN: 1* 10M/100M Port
* USB: 1* USB2.0
* SD: 1* MicroSD socket
* UART: 1* TTL, Baudrate 57600

Descriptions:
  Previous supported device YOUKU yk1 is actually Youku YK-L1. Though they look
  really different, the only hardware difference between the two models is flash
  size, YK-L1 has 32 MiB flash but YK-L1c has 16MiB. It seems that YK-L1c can
  compatible with YK-L1's firmware but it's better to split it to different models.

  It is easy to identify the models by looking at the label on the bottom of the
  device. The label has the model number "YK-L1" or "YK-L1c". Due to different flash
  sizes, YK-L1c that using previous YK-L1's firmware needs to apply "force update"
  to install compatible firmware, so please backup config file before system upgrade.

Signed-off-by: Shiji Yang <yangshiji66@qq.com>

DragonBluep commented:
> do the 2 devices have the same FCC ID or different? this would be good to include in commit message, to help identify differences.

I did not find FCC ID on the device label but there is model number "YK-L1" or "YK-L1c" on the bottom of the device. I just included it in the commit message.

adschm commented:
This looks good, thanks for keeping up.
I will merge it during the weekend.
